### DISCUSSION TITLE
Error while when i hit the login enpoint

### DISCUSSION THREAD

**Discussion by patrickkabwe** [gladly-team/next-firebase-auth#182] [Q&A]
"Firebase ID token has invalid signature. See https://firebase.google.com/docs/auth/admin/verify-id-tokens for details on how to retrieve an ID token."

import { setAuthCookies } from "next-firebase-auth";
import initAuth from "../../../firebase/initAuth"; // the module you created above

initAuth();

export default async (req, res) => {
  try {
    await setAuthCookies(req, res);

  } catch (error) {
    res.status(400).send({ error: error.message, success: false });
  }
};

Please HELP

**Comment by kmjennison** [comment]
It's hard to tell what's going wrong here without a clear way to reproduce it, so please share more info or a reproduction, if possible.

Here are a few things that might point you in the right direction:
* Double-check your Firebase config values are correct. See [related docs on troubleshooting](https://github.com/gladly-team/next-firebase-auth#i-get-an-invalid_custom_token-error-when-trying-to-get-a-refresh-token).
* Are you manually printing and verifying the token? Some people ran into this issue when copy-pasting a truncated token: https://github.com/flutter/flutter/issues/31635, https://github.com/firebase/firebase-cpp-sdk/issues/3
* Maybe related to #174: there are some errors this module doesn't handle but probably should.
* You could try logging and inspecting the JWT to try to identify what's wrong with its structure.

Hope this helps. Please follow up if you figure out what's going on.
